src/internal/common/idname/idname.go

```diff
@@ -0,0 +1,89 @@
+package idname
+
+import (
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type Provider interface {
+	// the canonical id of the thing, generated and usable
+	//  by whichever system has ownership of it.
+	ID() string
+	// the human-readable name of the thing.
+	Name() string
+}
+
+var _ Provider = &Is{}
+
+// Is provides an id-name tuple.
+type Is struct {
+	IDV   string
+	NameV string
+}
+
+func (is Is) ID() string   { return is.IDV }
+func (is Is) Name() string { return is.NameV }
+
+type Cacher interface {
+	IDOf(name string) (string, bool)
+	NameOf(id string) (string, bool)
+	IDs() []string
+	Names() []string
+	ProviderForID(id string) Provider
+	ProviderForName(id string) Provider
+}
+
+var _ Cacher = &Cache{}
+
+// Cache holds a cache of id-name mappings.
+type Cache struct {
+	IDToName map[string]string
+	NameToID map[string]string
+}
```

Comment:
I'd really feel more comfortable if we didn't export this or any of its fields

Reply:
sure, can do.  Forgot I'd meant to give that a constructor.